ext/pdo/pdo_dbh.c

```diff
@@ -38,6 +38,51 @@
 
 static int pdo_dbh_attribute_set(pdo_dbh_t *dbh, long attr, zval *value TSRMLS_DC);
 
+PDO_API zval *pdo_throw_exception_ex(zend_class_entry *exception_ce, pdo_error_type *pdo_err, long code TSRMLS_DC, const char *format, ...) /* {{{ */
```

Comment:
Every time pdo_throw_exception_ex is ever called exception_ce parameter is always php_pdo_get_exception(). Since this is clearly a PDO-specific function, maybe this parameter should be eliminated and this function should call php_pdo_get_exception() instead?